kubelet/tests/test_kubelet.py

```diff
@@ -763,3 +771,23 @@ def test_kubelet_stats_summary_not_available(monkeypatch, aggregator, tagger):
 
     check.check(instance)
     check._retrieve_stats.assert_called_once()
+
+
+def test_system_container_metrics(monkeypatch, aggregator, tagger):
+    check = KubeletCheck('kubelet', None, {}, [{}])
+    monkeypatch.setattr(
+        check, '_retrieve_stats', mock.Mock(return_value=json.loads(mock_from_file('stats_summary.json')))
+    )
+    monkeypatch.setattr(check, 'gauge', mock.Mock())
+
+    stats = check._retrieve_stats()
+    tags = ["instance:tag"]
+    check._report_system_container_metrics(stats, tags)
+
+    calls = [
+        mock.call('kubernetes.kubelet.cpu.usage', 36755862.0, tags),
+        mock.call('kubernetes.runtime.cpu.usage', 19442853.0, tags),
+        mock.call('kubernetes.runtime.memory.rss', 101273600.0, tags),
+        mock.call('kubernetes.kubelet.memory.rss', 88477696.0, tags),
+    ]
+    check.gauge.assert_has_calls(calls, any_order=True)
```

Comment:
You have the `aggregator` fixture just for this use case. No need to mock `gauge` and you can run `aggregator.assert_metric` on those.

Reply:
thanks!